homeassistant/components/reolink/media_source.py

```diff
@@ -0,0 +1,321 @@
+                title="High resolution",
+                can_play=False,
+                can_expand=True,
+            ),
+        ]
+
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"RESs+{config_entry_id}+{channel}",
+            media_class=MediaClass.CHANNEL,
+            media_content_type=MediaType.PLAYLIST,
+            title=host.api.camera_name(channel),
+            can_play=False,
+            can_expand=True,
+            children=children,
+        )
+
+    async def _generate_camera_days(
+        self, config_entry_id: str, channel: int, stream: str
+    ) -> BrowseMediaSource:
+        """Return all days on which recordings are available for a reolink camera."""
+        host = self.data[config_entry_id].host
+
+        # We want today of the camera, not necessarily today of the server
+        now = host.api.time()
+        if not now:
+            now = await host.api.async_get_time()
+        start = now - dt.timedelta(days=31)
+        end = now
+
+        children = []
+        _LOGGER.debug(
+            "Requesting recording days of %s from %s to %s",
+            host.api.camera_name(channel),
+            start,
+            end,
+        )
+        (statuses, _) = await host.api.request_vod_files(
+            channel, start, end, status_only=True, stream=stream
+        )
+        for status in statuses:
+            for day in status.days:
+                children.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"DAY+{config_entry_id}+{channel}+{stream}+{status.year}+{status.month}+{day}",
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaType.PLAYLIST,
+                        title=f"{status.year}/{status.month}/{day}",
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+
+        res_name = "High res." if stream == "main" else "Low res."
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=f"DAYS+{config_entry_id}+{channel}+{stream}",
+            media_class=MediaClass.CHANNEL,
+            media_content_type=MediaType.PLAYLIST,
+            title=f"{host.api.camera_name(channel)} {res_name}",
+            can_play=False,
+            can_expand=True,
+            children=children,
+        )
+
+    async def _generate_camera_files(
+        self,
+        config_entry_id: str,
+        channel: int,
+        stream: str,
+        year: int,
+        month: int,
+        day: int,
+    ) -> BrowseMediaSource:
+        """Return all recording files on a specific day of a Reolink camera."""
+        host = self.data[config_entry_id].host
+
+        start = dt.datetime(year, month, day, hour=0, minute=0, second=0)
+        end = dt.datetime(year, month, day, hour=23, minute=59, second=59)
```

Comment:
I think the HA dt_util has functions to get the start of the day (and maybe the end of the day?)

Reply:
it does, but it requires a date/datetime already and since there are only y/m/d vars here, it wouldn't save any steps or be that much cleaner to write. That method is more of a helper for when you have a date or datetime (or nothing) and want the SOD. Also, we ant to avoid setting the tzinfo, since we want the date/time to be relative to the camera/device and not "adjusted" to the camera.